### PR TITLE
Add Unbounded knapsack in C code

### PR DESCRIPTION
**Pull request by Zuoxun**
Running results are as follows:
> 不超过背包容量的最大物品价值为 22
不超过背包容量的最大物品价值为 22

If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/c/chapter_dynamic_programming/unbounded_knapsack.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+#include "../utils/common.h"
+
+/* 求最大值 */
+int max(int a, int b) {
+    return a > b ? a : b;
+}
+
+/* 完全背包：动态规划 */
+int unboundedKnapsackDP(int wgt[], int val[], int cap, int wgt_size) {
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/unbounded_knapsack.c:15]: `wgtSize`

File: codes/c/chapter_dynamic_programming/unbounded_knapsack.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                // 不选和选物品 i 这两种方案的较大值
+                dp[i][c] = max(dp[i - 1][c], dp[i][c - wgt[i - 1]] + val[i - 1]);
+            }
+        }
+    }
+    return dp[n][cap];
+}
+
+/* 完全背包：空间优化后的动态规划 */
+int unboundedKnapsackDPComp(int wgt[], int val[], int cap, int wgt_size) {
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/unbounded_knapsack.c:36]: Same as above

File: codes/c/chapter_dynamic_programming/unbounded_knapsack.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        }
+    }
+    return dp[cap];
+}
+
+/* Driver code */
+int main() {
+    int wgt[] = {1, 2, 3};
+    int val[] = {5, 11, 15};
+    int wgt_size = sizeof(wgt) / sizeof(wgt[0]);
```
Comment (Gonglja) [inline codes/c/chapter_dynamic_programming/unbounded_knapsack.c:60]: Same as above

File: codes/c/chapter_dynamic_programming/unbounded_knapsack.c (hunk 1 above)
Comment (Zuoxun) [reply codes/c/chapter_dynamic_programming/unbounded_knapsack.c]: I've changed these naming methods in all my PRs, thx for your review. @Gonglja

**Review by Gonglja — COMMENTED** [review]
Ok, thanks @Zuoxun 
Can be merged @krahets

**Review by krahets — APPROVED** [review]

**Pull request merged**